src/Kafka/KafkaMessageBrowser/components/DateTimePicker.tsx

```diff
@@ -23,17 +25,30 @@ export const DateTimePicker: VoidFunctionComponent<DateTimePickerProps> = ({
     }
   };
 
-  const onSelectTime: TimePickerProps["onChange"] = (_, hour, minute) => {
-    //onChange();
-    if (date) {
+  const onSelectTime: TimePickerProps["onChange"] = (
+    time,
+    hour,
+    minute,
+    __,
+    isValid
+  ) => {
+    if (
+      isValid &&
+      date &&
+      hour != undefined &&
+      hour > 0 &&
+      (time.includes("AM") || time.includes("PM"))
+    ) {
       let newDate = date;
       if (hour !== undefined) {
         newDate = setHours(newDate, hour);
       }
       if (minute !== undefined) {
         newDate = setMinutes(newDate, minute);
       }
+
       onChange(formatISO(newDate));
+      setIsTimeSelected(true);
```

Comment:
When is this set back to `false`?

Reply:
We would not want to set this to false, we only want this to be false when the user has not interacted with the time picker component so that he can see the placeholder rather than the default time 0:00AM for the date he has selected. Once the user starts typing or selects a value, we will no longer need the placeholder.
We use this to prevent 0:00AM to show up after we select a date as that is an invalid format. We want the initial value to be a placeholder